### DISCUSSION TITLE
How to write a show-rule that matches inter-word whitespace?

### DISCUSSION THREAD

**Discussion by t-rapp** [typst/typst#3851] [Q&A] [answered]
I have defined a simple show rule which replaces the word "page" and all following whitespace with its abbreviation "pg." plus a non-breakable space character.

```
#set page(width: 6cm, height: auto)

#show regex("page\s"): [pg.] + sym.space.nobreak

// Ok: Doesn't match the showrule
pages

page.

// Ok: Matches the showrule
page 1

page 1-4

// Err: Should match the showrule?
page ?

page  2

page $2+3$
```

![Check ShowRuleMatchWord 01](https://github.com/typst/typst/assets/20061583/367e40a3-393b-4f97-909e-f726a7556b83)

It seems that the show-rule is not applied when the character following the rule is a punctuation character, another whitespace character, or when changing into math mode.

Is it possible to write a show-rule that matches more than a single word? How could I adjust the pattern in the example to match some of the cases at the bottom?

**Comment by laurmaedje** [comment, marked as answer]
This is unfortunately a current limitation, see https://github.com/typst/typst/issues/86

**Reply by t-rapp** [reply]
Thanks for the pointer. Will follow that ticket for future updates.
